packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionsTable/TransactionsTable.tsx

```diff
@@ -62,6 +70,73 @@ export const TransactionDateTime = ({
   )
 }
 
+export const CustomAddressTxExplorer = ({
+  tx,
+  explorerClassName = 'arb-hover underline'
+}: {
+  tx: MergedTransaction
+  explorerClassName?: string
+}) => {
+  const { address } = useAccount()
+  const { l1, l2 } = useNetworksAndSigners()
+
+  const isCustomSenderTx = useMemo(() => {
+    if (!tx.sender || !address) {
+      return false
+    }
+    return tx.sender.toLowerCase() !== address.toLowerCase()
+  }, [tx.sender, address])
+
+  const isCustomDestinationTx = useMemo(() => {
+    if (!address || !tx.destination) {
+      return false
+    }
+    return tx.destination.toLowerCase() !== address.toLowerCase()
+  }, [tx.destination, address])
+
+  const explorerChainId = useMemo(() => {
+    if (!isCustomSenderTx && !isCustomDestinationTx) {
+      return null
+    }
+    if (tx.isWithdrawal) {
+      if (isCustomSenderTx) {
+        return l2.network.id
+      }
+      return l1.network.id
+    }
+    if (isCustomSenderTx) {
+      return l1.network.id
+    }
+    return l2.network.id
```

Comment:
```suggestion
    if (tx.isWithdrawal) {
      // this is a withdrawal, so
      // if it's a different sender, show their L2 address (where the withdrawal originated)
      // otherwise it's a custom destination, show their L1 address (where the funds will land)
      return isCustomSenderTx ? l2.network.id : l1.network.id
    }
    
    // this is a deposit, so
    // if it's a different sender, show their L1 address (where the deposit originated)
    // otherwise it's a custom destination, show their L2 address (where the funds will land)
    return isCustomSenderTx ? l1.network.id : l2.network.id
```